esp/esdllib/esdl_script.cpp

```diff
@@ -328,42 +666,86 @@ class CEsdlTransformOperationSetValue : public CEsdlTransformOperationBase
         }
         return m_target.str();
     }
-    virtual bool doSet(IXpathContext * xpathContext, IPropertyTree *tree, const char *value)
+    virtual bool doSet(IXpathContext * sourceContext, IXpathContext *targetContext, const char *value)
     {
         StringBuffer xpath;
-        const char *target = getTargetPath(xpathContext, xpath);
-        ensurePTree(tree, target);
-        tree->setProp(target, value);
+        const char *target = getTargetPath(sourceContext, xpath);
+        targetContext->ensureSetValue(target, value, m_required);
         return true;
     }
 };
 
-class CEsdlTransformOperationRenameNode : public CEsdlTransformOperationBase
+class CEsdlTransformOperationNamespace : public CEsdlTransformOperationWithoutChildren
+{
+protected:
+    StringAttr m_prefix;
+    StringAttr m_uri;
+    bool m_current = false;
+
+public:
+    CEsdlTransformOperationNamespace(XmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
+    {
+        if (m_traceName.isEmpty())
+            m_traceName.set(stag.getValue("name"));
+
+        const char *pfx = stag.getValue("prefix");
+        const char *uri = stag.getValue("uri");
+        if (!pfx && isEmptyString(uri))
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname.str(), "without prefix or uri", m_traceName.str(), !m_ignoreCodingErrors);
+        m_uri.set(uri);
+        m_prefix.set(pfx);
+        m_current = getStartTagValueBool(stag, "current", m_uri.isEmpty());
+    }
+
+    virtual void toDBGLog() override
+    {
+#if defined(_DEBUG)
+        DBGLOG(">%s> %s(prefix('%s'), uri('%s'), current(%d))", m_traceName.str(), m_tagname.str(), m_prefix.str(), m_uri.str(), m_current);
+#endif
+    }
+
+    virtual ~CEsdlTransformOperationNamespace(){}
+
+    virtual bool process(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
+    {
+        targetContext->setLocationNamespace(m_prefix, m_uri, m_current);
+        return false;
+    }
+};
+
+class CEsdlTransformOperationRenameNode : public CEsdlTransformOperationWithoutChildren
 {
 protected:
     StringAttr m_target;
     StringAttr m_new_name;
     Owned<ICompiledXpath> m_xpath_target;
     Owned<ICompiledXpath> m_xpath_new_name;
+    bool all = false;
 
 public:
-    CEsdlTransformOperationRenameNode(IPropertyTree *tree, const StringBuffer &prefix) : CEsdlTransformOperationBase(tree, prefix)
+    CEsdlTransformOperationRenameNode(XmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
     {
-        if (isEmptyString(tree->queryProp("@new_name")) && isEmptyString(tree->queryProp("@xpath_new_name")))
+        const char *new_name = stag.getValue("new_name");
+        const char *xpath_new_name = stag.getValue("xpath_new_name");
+        if (isEmptyString(new_name) && isEmptyString(xpath_new_name))
```

Comment:
Perhaps set m_traceName to new_name or xpath_new_name if empty?

Reply:
will do.